perun-core/src/main/java/cz/metacentrum/perun/core/implApi/AttributesManagerImplApi.java

```diff
@@ -213,6 +213,18 @@
 	List<Attribute> getAttributes(PerunSession sess, Member member, Resource resource, List<String> attrNames) throws InternalErrorException;
 
 	/**
+	 * Get all attributes (empty and virtual too) associated with the member on the resource which have name in list attrNames.
```

Comment:
Word 'member' replaced for 'group'.
I think you get all attributes (non-empty, empty, virtual) because you have specific list of names.